packages/office-ui-fabric-react/src/components/TextField/TextField.tsx

```diff
@@ -131,7 +133,7 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
       this._latestValue = newProps.value;
       this.setState(
         {
-          value: newProps.value,
+          value: newProps.value || '',
```

Comment:
If value is `0` numerical, this would not work.

Reply:
I don't quite understand why this one is needed tho.

Reply:
Value is defined as a string or undefined only, not as a number. This is needed to bring symmetry with state behavior, which otherwise defines state.value to '' if value prop is undefined (see constructor). I made this change because I found that if I changed value prop to undefined, then state value became undefined, which is inconsistent with initial state in constructor.

Reply:
ok please merge if you're good with everything!